src/ess/reduce/time_of_flight/interpolator_numba.py

```diff
@@ -0,0 +1,156 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2025 Scipp contributors (https://github.com/scipp)
+import numpy as np
+from numba import njit, prange
+
+
+@njit(boundscheck=False, cache=True, fastmath=True, parallel=True)
```

Comment:
Are you sure about `boundscheck=False`? There is no check in this function that the arrays are all the same size. Is this needed for performance?

Reply:
It was for perfomance, but we can put it back if you like. I can also disable `fastmath` as @SimonHeybrock requested above.